src/index.js

```diff
@@ -1098,42 +1098,36 @@ export function addAction (name, opts) {
     opts.response = opts.response || function (response) { return response }
     opts.responseError = opts.responseError || function (err) { return utils.reject(err) }
     mapper[name] = function (id, _opts) {
-      if (utils.isObject(id)) {
-        _opts = id
-      }
       _opts = _opts || {}
-      let adapter = this.getAdapter(opts.adapter || this.defaultAdapter || 'http')
-      let config = {}
-      utils.fillIn(config, opts)
-      if (!_opts.hasOwnProperty('endpoint') && config.endpoint) {
-        _opts.endpoint = config.endpoint
-      }
+      utils.fillIn(_opts, opts)
```

Comment:
Javascript is pretty dynamic-y but it may be worth considering having a limited `fillIn` in utils e.g. `fillIn(dest, src, [<selected keys>])` so that the attributes merged from one object to another can be explicitly defined.